experimental/FTheoryTools/src/types.jl

```diff
@@ -0,0 +1,107 @@
+################################################
+# 1: The Julia types for FTheoryTools
+################################################
+
+abstract type AbstractFTheoryModel end
+
+ToricOrNonToricCoveredScheme = Union{ToricCoveredScheme{BRT}, CoveredScheme{BRT}} where {BRT<:Ring}
+
+@attributes mutable struct ClosedSubschemeModel <: AbstractFTheoryModel
```

Comment:
Here, and probably already for the abstract supertypes, I would introduce type parameters which already allow for the identification of the type of, say, the base space, etc. But, since time is pressing this week, we can also do that later.